### ISSUE TITLE
Missing italic styles in 0.11?

### ISSUE THREAD

**Issue by brc-dd** [shikijs/shiki#343]
Reproduction (https://runkit.com/brc-dd/6309d1529e302e000a926cad):

```js
const shiki_10 = require('shiki@0.10.1')
const shiki_11 = require('shiki')

const h_10 = await shiki_10.getHighlighter({ theme: 'material-palenight' })
const h_11 = await shiki_11.getHighlighter({ theme: 'material-palenight' })

console.log(h_10.codeToHtml(`import foo from 'bar'`, { lang: 'js' }))
console.log(h_11.codeToHtml(`import foo from 'bar'`, { lang: 'js' }))
```

Output (notice the difference in style of `import` and `from`):

![image](https://user-images.githubusercontent.com/40380293/187021606-dded8afe-3a1a-4ced-89f0-edacf7cdab35.png)

Is some additional config required to render it like before, or the theme itself has changed?

**Comment by orta** [comment]
yeah, I agree it's likely that the theme or langauge updated inbetween

**Comment by brc-dd** [comment]
Theme still has that italics stuff. Say here are these things for comments. But if you try to render a comment, then it won't have any font-style. Also, if the language had changed, then how are colors being properly applied? Seems like some bug to me.

https://github.com/shikijs/shiki/blob/525e986e8f5d96ed3fbefb78de157e4d612733a5/packages/shiki/themes/material-palenight.json#L104-L119

**Comment by orta** [comment]
Yeah, I agree, though [not much changed inside the renderer](https://github.com/shikijs/shiki/compare/v0.10.1...v0.11.0#diff-919ab74d473366fabe3f5efd32806e5e8af7202cc565a46a71bb57bb303eb33eR1) - and it still looks like the italics support is in there. 

Open to some tests in the area to see if there's something which broke

**Comment by brc-dd** [comment]
Just a question. why is there bitwise and instead of logical and everywhere?

![image](https://user-images.githubusercontent.com/40380293/187085638-96b7a6d4-f18f-442f-9da4-ea3453b6c67d.png)

**Comment by LaPenna** [comment]
Experiencing the same lack of italics issue.

Interestingly, if I specify `"fontStyle": "bold"` in the theme, then it applies the `"font-style: italic"` style.

Here are some tests

```js
import { getHighlighter } from '../index'

// passes
test('No font-style is applied if none is specified', async () => {
  const highlighter = await getHighlighter({
    theme: 'font-none',
    langs: ['js']
  })
  const out = highlighter.codeToHtml(`function() {return 1}`, { lang: 'js' })

  expect(out).not.toContain('font-style:')
})

// fails - doesn't apply any font-style
test('An italic font-style is applied if specified', async () => {
  const highlighter = await getHighlighter({
    theme: 'font-italic',
    langs: ['js']
  })
  const out = highlighter.codeToHtml(`function() {return 1}`, { lang: 'js' })

  expect(out).toContain('font-style: italic')
})

// fails - incorrectly applies font-style: italic
test('A bold font-style is applied if specified', async () => {
  const highlighter = await getHighlighter({
    theme: 'font-bold',
    langs: ['js']
  })
  const out = highlighter.codeToHtml(`function() {return 1}`, { lang: 'js' })

  expect(out).toContain('font-style: bold')
})

// fails - test for the bug encountered above
test('A bold fontStyle should not apply an italic font-style', async () => {
  const highlighter = await getHighlighter({
    theme: 'font-bold',
    langs: ['js']
  })
  const out = highlighter.codeToHtml(`function() {return 1}`, { lang: 'js' })

  expect(out).not.toContain('font-style: italic')
})
```

For simplicity & isolation I used the following stubs as the themes but the same issue occurs when I test with any normal theme that uses italics.

```js
// font-none.json
{
  "tokenColors": [
    {
      "name": "Keyword",
      "scope": "keyword",
      "settings": {
        "foreground": "#ffffff"
      }
    }
  ]
}

// font-bold.json
{
  "tokenColors": [
    {
      "name": "Keyword",
      "scope": "keyword",
      "settings": {
        "foreground": "#ffffff",
        "fontStyle": "bold"
      }
    }
  ]
}

// font-italic.json
{
  "tokenColors": [
    {
      "name": "Keyword",
      "scope": "keyword",
      "settings": {
        "foreground": "#ffffff",
        "fontStyle": "italic"
      }
    }
  ]
}
```

**Comment by octref** [comment]
Thanks @LaPenna for the test case! This is a tokenization problem and I fixed it by upgrading to `vscode-textmate@7.0.1`.
